src/Features/Core/Portable/Diagnostics/AbstractHostDiagnosticUpdateSource.cs

```diff
@@ -1,119 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System;
-using System.Collections.Immutable;
-using System.Linq;
-using Microsoft.CodeAnalysis.Shared.Collections;
-using Roslyn.Utilities;
-
-namespace Microsoft.CodeAnalysis.Diagnostics;
-
-/// <summary>
-/// Diagnostic update source for reporting workspace host specific diagnostics,
-/// which may not be related to any given project/document in the solution.
-/// For example, these include diagnostics generated for exceptions from third party analyzers.
-/// </summary>
-internal abstract class AbstractHostDiagnosticUpdateSource
-{
-    private ImmutableDictionary<DiagnosticAnalyzer, ImmutableHashSet<DiagnosticData>> _analyzerHostDiagnosticsMap =
-        ImmutableDictionary<DiagnosticAnalyzer, ImmutableHashSet<DiagnosticData>>.Empty;
-
-    public abstract Workspace Workspace { get; }
-
-    public event EventHandler<ImmutableArray<DiagnosticsUpdatedArgs>>? DiagnosticsUpdated;
-
-    public void RaiseDiagnosticsUpdated(ImmutableArray<DiagnosticsUpdatedArgs> args)
-    {
-        if (!args.IsEmpty)
-            DiagnosticsUpdated?.Invoke(this, args);
-    }
-
-    public void ClearAnalyzerReferenceDiagnostics(AnalyzerFileReference analyzerReference, string language, ProjectId projectId)
-    {
-        // Perf: if we don't have any diagnostics at all, just return right away; this avoids loading the analyzers
-        // which may have not been loaded if you didn't do too much in your session.
-        if (_analyzerHostDiagnosticsMap.Count == 0)
-            return;
-
-        using var argsBuilder = TemporaryArray<DiagnosticsUpdatedArgs>.Empty;
-        var analyzers = analyzerReference.GetAnalyzers(language);
-        AddArgsToClearAnalyzerDiagnostics(ref argsBuilder.AsRef(), analyzers, projectId);
-        RaiseDiagnosticsUpdated(argsBuilder.ToImmutableAndClear());
```

Comment:
this was a call to raise hte event.  no more event, then no more method.  once tehre's no more method, there's no need to do thingsl ike add arguments to builders anymore.